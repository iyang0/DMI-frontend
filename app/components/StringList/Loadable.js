/**
 *
 * Asynchronously loads the component for StringList
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
