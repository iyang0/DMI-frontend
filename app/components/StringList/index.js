/**
 *
 * StringList
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import LoadingIndicator from 'components/LoadingIndicator';
import List from 'components/List';
import ListItem from 'components/ListItem';
import Empty from 'components/Empty';
// import styled from 'styled-components';

function StringList({ loading, error, list }) {
  if (loading) {
    return <List component={LoadingIndicator} />;
  }

  if (error !== false) {
    const ErrorComponent = () => (
      <ListItem item="Something went wrong, please try again!" />
    );
    return <List component={ErrorComponent} />;
  }

  if (list !== false) {
    return <List items={list} component={Empty} />;
  }

  return null;
}

StringList.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.any,
  list: PropTypes.any,
};

export default memo(StringList);
