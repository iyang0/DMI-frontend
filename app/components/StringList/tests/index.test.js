/**
 *
 * Tests for StringList
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import StringList from '../index';

describe('<StringList />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<StringList />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render the loading indicator when its loading', () => {
    const { container } = render(<StringList loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing is provided', () => {
    const { container } = render(
      <StringList list={false} error={false} loading={false} />,
    );

    expect(container.firstChild).toBeNull();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<StringList />);
    expect(firstChild).toMatchSnapshot();
  });
});
