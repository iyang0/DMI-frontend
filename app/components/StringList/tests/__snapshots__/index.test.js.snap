// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringList /> Should render and match the snapshot 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 laCwDT"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <li
      class="Wrapper-euo0oy-0 jLvxtc"
    >
      <div
        class="Item-sc-3y9mie-0 fKXSkT"
      >
        Something went wrong, please try again!
      </div>
    </li>
  </ul>
</div>
`;
